pkg/build/builder/source_test.go

```diff
@@ -0,0 +1,26 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestCheckRemoteGit(t *testing.T) {
+	var err error
+	err = checkRemoteGit("https://github.com/openshift/private-repo.git", 10*time.Second)
```

Comment:
can we mock this with a httptest server that always returns 401 unauthenticated?